src/phylum/init/cli.py

```diff
@@ -45,23 +45,61 @@
 }
 
 TOKEN_ENVVAR_NAME = "PHYLUM_TOKEN"
-PHYLUM_PATH = pathlib.Path.home() / ".phylum"
-PHYLUM_BIN_PATH = PHYLUM_PATH / "phylum"
-SETTINGS_YAML_PATH = PHYLUM_PATH / "settings.yaml"
 
-# Potential features to add:
-#  * Add logging support, a verbosity option to control it, and swap out print statements for logging
-#  * Check for valid versions by using the GitHub API to compare against actual releases
-#    * If so, also programmatically generate the SUPPORTED_TARGET_TRIPLES
-#  * Use the GitHub API ("https://api.github.com") to more programmatically get the archive URL
-#  * Add a `--list` option, to show which versions are available
-#  * Add an option to account for pre-releases
+
+def get_phylum_settings_path(version):
+    """Get the Phylum settings path based on a provided version."""
+    home_dir = pathlib.Path.home()
+    version = version_check(version)
+
+    config_home_path = os.getenv("XDG_CONFIG_HOME")
+    if not config_home_path:
+        config_home_path = home_dir / ".config"
+    phylum_config_path = pathlib.Path(config_home_path) / "phylum" / "settings.yaml"
+
+    # The Phylum config path changed following the v2.2.0 release, to adhere to the XDG Base Directory Spec
+    # Reference: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+    if Version(canonicalize_version(version)) <= Version("v2.2.0"):
+        phylum_config_path = home_dir / ".phylum" / "settings.yaml"
+
+    return phylum_config_path
+
+
+def get_phylum_bin_path(version):
+    """Get the path to the Phylum binary based on a provided version."""
+    home_dir = pathlib.Path.home()
+    version = version_check(version)
+
+    # The Phylum binary path changed following the v2.2.0 release, to adhere to the XDG Base Directory Spec
+    # Reference: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+    phylum_bin_path = home_dir / ".local" / "bin" / "phylum"
+
+    if Version(canonicalize_version(version)) <= Version("v2.2.0"):
+        phylum_bin_path = home_dir / ".phylum" / "phylum"
+
+    return phylum_bin_path
+
+
+def get_latest_version():
+    """Get the "latest" version programmatically and return it."""
+    # API Reference: https://docs.github.com/en/rest/releases/releases#get-the-latest-release
+    github_api_url = "https://api.github.com/repos/phylum-dev/cli/releases/latest"
+
+    req = requests.get(github_api_url, timeout=5.0)
+    req.raise_for_status()
+    req_json = req.json()
+
+    # The "name" entry stores the GitHub Release name, which could be set to something other than the version.
+    # Using the "tag_name" entry is better since the tags are much more tightly coupled with the release version.
+    latest_version = req_json.get("tag_name")
```

Comment:
🤔 For everything we've got, `name` and `tag_name` should be the same, so either one should work. But what really matters here is what this is used for, which is the download URL:

```
https://github.com/phylum-dev/cli/releases/download/{version}/{artifact}
```

Does that `{version}` correspond to `tag_name` or `name`? I would assume they'd use `name` there, but maybe we should test that.

Reply:
It looks like GitHub releases are referenced by `tag_name` and not `name` (which is the release name). I tried this out by changing the release name of `v2.2.0-rc1` to `release-v2.2.0-rc1-title` and witnessed the changes in the [API GET for releases](https://api.github.com/repos/phylum-dev/cli/releases): 

```
...
    "tag_name": "v2.2.0-rc1",
    "target_commitish": "main",
    "name": "release-v2.2.0-rc1-title",
    "draft": false,
    "prerelease": true,
    "created_at": "2022-04-19T14:40:34Z",
    "published_at": "2022-04-19T14:54:46Z",
    "assets": [
      {
        "url": "https://api.github.com/repos/phylum-dev/cli/releases/assets/63007509",
        "id": 63007509,
        "node_id": "RA_kwDOFNQrN84DwWsV",
        "name": "phylum-aarch64-apple-darwin.zip",
        "label": "",
...
        "created_at": "2022-04-19T14:54:46Z",
        "updated_at": "2022-04-19T14:54:47Z",
        "browser_download_url": "https://github.com/phylum-dev/cli/releases/download/v2.2.0-rc1/phylum-aarch64-apple-darwin.zip"
      },
...
```

Notice the name shows as `release-v2.2.0-rc1-title` but the `tag_name` was unchanged. The important part is that the asset `browser_download_url`s are still referenced by `tag_name`:

```
"browser_download_url": "https://github.com/phylum-dev/cli/releases/download/v2.2.0-rc1/phylum-aarch64-apple-darwin.zip"
```

This tracks with the API documents, since they show how to “Get a release by tag name” - https://docs.github.com/en/rest/releases/releases#get-a-release-by-tag-name - but not by release name.